src/ray/gcs/redis_context.cc

```diff
@@ -279,9 +316,13 @@ RedisContext::~RedisContext() {
   }
 }
 
+void RedisContext::ResetSyncContext() { context_.reset(); }
+
+void RedisContext::ResetAsyncContext() { redis_async_context_.reset(); }
+
 void RedisContext::Disconnect() {
-  context_.reset();
-  redis_async_context_.reset();
+  ResetSyncContext();
+  ResetAsyncContext();
```

Comment:
Actually this can cause nullptr exception since there might be some `RedisRequestContext` that hold the raw pointer to redis async context?

Reply:
Moved nullptr check for redis async context to `RedisContext::async_context` in this commit. `RedisRequestContext` always calls `RedisContext::async_context` to get the redis async context so this can prevent nullptr exception.

https://github.com/ray-project/ray/pull/48781/commits/9779e4c05bde8a454f40f2e64917b3274780cfde